fiat-sql/src/main/kotlin/com/netflix/spinnaker/fiat/permissions/SqlPermissionsRepository.kt

```diff
@@ -118,23 +119,46 @@ class SqlPermissionsRepository(
         if (UNRESTRICTED_USERNAME == id) {
             return Optional.of(unrestricted)
         }
+        return getUserPermissions(id, unrestricted)
+    }
+
+    private fun getUserPermissions(id: String, unrestrictedUser: UserPermission?): Optional<UserPermission> {
+        // Check if the user exists and if they do are they an admin
+        val isAdmin = withRetry(RetryCategory.READ) {
+            jooq.select(USER.ADMIN)
+                .from(USER)
+                .where(USER.ID.eq(id))
+                .fetchOne(USER.ADMIN)
+        } ?: return Optional.empty()
```

Comment:
When would this be null? The withRetry call below doesn't have the null check.

Reply:
When the user doesn't exist in the Fiat permission repository. The logic here should match the RedisPermissionRepository.